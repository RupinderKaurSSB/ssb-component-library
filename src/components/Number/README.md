Number
========

> A component for showing number in different size

### Usage

#### HTML

```html
<div class="ssb-number small">123</div>

<div class="ssb-number medium">123 456</div>

<div class="ssb-number large">123 456 789</div>
```

#### React

```jsx harmony
<Number number="123" size="small" />

<Number number="123456" size="medium" />

<Number number="123456789" size="large" />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| size | 'small', 'medium' or 'large' | Change size on number|
| number | Required 'string' or 'number' | Number content |
