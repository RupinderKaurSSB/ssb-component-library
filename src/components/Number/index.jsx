import React from 'react';
import PropTypes from 'prop-types';

const Number = ({
	children, size,
}) => (
	<div className={`ssb-number ${size}`}>
		{children}
	</div>
);

Number.defaultProps = {
	size: 'small',
};

Number.propTypes = {
	children: PropTypes.node,
	size: PropTypes.oneOf(['small', 'medium', 'large']),
};

export default Number;
