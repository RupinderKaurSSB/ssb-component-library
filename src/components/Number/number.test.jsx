import React from 'react';
import {shallow} from 'enzyme';
import Number from './index';

describe('Number component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Number>599 246 391</Number>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<Number size="small">599 246 391</Number>);
		expect(wrapper.hasClass('small')).toEqual(true);
	});

});
