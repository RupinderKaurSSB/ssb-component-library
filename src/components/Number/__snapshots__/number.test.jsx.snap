// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component Matches the snapshot 1`] = `
<div
  className="ssb-number small"
>
  59 24 391
</div>
`;
